src/ml/train/focus_classifiers.py

```diff
@@ -4,6 +4,7 @@
 import os
 import time
 import pickle
+import copy
 
 import numpy as np
```

Comment:
Just a comment about imports lines 12,14,15 and most of 20,21 are unnecessary imports.